extensions/crypto/00_webidl.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const webidl = window.__bootstrap.webidl;
+  webidl.converters["AlgorithmIdentifier"] = (V, _opts) => {
+    if (typeof V == "string") {
+      return webidl.converters["DOMString"](V);
+    }
+
+    return webidl.converters["Algorithm"](V);
```

Comment:
Pass opts to the converters to maintain context, add the comment for which IDL union this implements, and please follow the spec algorithm (current impl is wrong): https://heycam.github.io/webidl/#es-union. Relevant here are steps 10.3 and 14.